analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/ComplianceAnalyzerPlugin.java

```diff
@@ -0,0 +1,235 @@
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.collect.Lists;
+import eu.fasten.core.plugins.DBConnector;
+import eu.fasten.core.plugins.KafkaPlugin;
+import io.kubernetes.client.openapi.ApiClient;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.Configuration;
+import io.kubernetes.client.openapi.apis.BatchV1Api;
+import io.kubernetes.client.openapi.apis.CoreV1Api;
+import io.kubernetes.client.openapi.models.V1ConfigMap;
+import io.kubernetes.client.openapi.models.V1Job;
+import io.kubernetes.client.openapi.models.V1Service;
+import io.kubernetes.client.util.Config;
+import io.kubernetes.client.util.KubeConfig;
+import io.kubernetes.client.util.Yaml;
+import org.jooq.DSLContext;
+import org.json.JSONObject;
+import org.pf4j.Extension;
+import org.pf4j.Plugin;
+import org.pf4j.PluginWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+
+/**
+ * Plugin which runs qmstr command line tool to detect
+ * license compatibility and compliance.
+ */
+public class ComplianceAnalyzerPlugin extends Plugin {
+
+    /**
+     * Kubernetes namespace where our demo objects are going to be deployed in.
+     */
+    protected static final String K8S_NAMESPACE = "default";
+
+    public ComplianceAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class CompliancePluginExtension implements KafkaPlugin {
+
+        private final Logger logger = LoggerFactory.getLogger(CompliancePluginExtension.class.getName());
+
+        protected String consumerTopic = "fasten.RepoCloner.out";
+        protected Throwable pluginError = null;
+        protected String repoUrl;
+
+        /**
+         * Path to the Kubernetes cluster credentials file
+         */
+        protected final String clusterCredentialsFilePath;
+
+        public CompliancePluginExtension(String clusterCredentialsFilePath) {
+            this.clusterCredentialsFilePath = clusterCredentialsFilePath;
+        }
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public void consume(String record) {
+            try { // Fasten error-handling guidelines
+
+                this.pluginError = null;
+                var consumedJson = new JSONObject(record);
+                if (consumedJson.has("input")) {
+                    consumedJson = consumedJson.getJSONObject("input");
```

Comment:
It is better not to go into `input` as the order of nested inputs can change depending on the order of plugins in the pipeline. Everything your plugin requires should be in the `payload`. If you need any attributes from RepoCloner's input, please create an [issue](https://github.com/fasten-project/fasten/issues) for that and I will implement it.

Reply:
Will do. But first, where can I find a schema of such Kafka Topic? I wanna make sure that the [`dummyKafkaTopic.json`](https://github.com/fasten-project/fasten/blob/endocode/compliancePlugin/analyzer/compliance-analyzer/dummyKafkaTopic.json) file has the right structure, and then request for a `repoUrl` field in the `payload`.

Reply:
Kafka topics are described in the Wiki. https://github.com/fasten-project/fasten/wiki/Kafka-Topics#fastenrepocloner
`repoUrl` is in the payload of `fasten.POMAnalyzer.out` so when you create the issue, I will adjust the RepoCloner to copy from the input to the output. Then it will be in the payload of `fasten.RepoCloner.out`

Reply:
Perfect, many thanks!

Reply:
Done! https://github.com/fasten-project/fasten/issues/71

Reply:
Implemented in eb738f6

Reply:
Fixed accordingly in dc9bc6fc.